### ISSUE TITLE
[enhancement] improve GUIEnhancements

### ISSUE THREAD

**Issue by lynxlynxlynx** [gemrb/gemrb#427]
Currently a bunch of ease-of-use extensions and similar are contingent on having GUIEnhancements defined and high enough (it's a bitfield). We set it by default in the config, so nobody even knows.

We need to revisit everything it toggles, since some things can just be made unconditional (eg. the last spell button for sorcerer spell selection), but more importantly, we need to document it better and make it easier to change. chilvence had sort of a gui toggler ready (I have the diff somewhere), but it wasn't finalized, so it's not in. And it was for bg2 only.

**Comment by bradallred** [comment]
there were a bunch of `TextArea` scrollbars that we will no longer need too. I think I already ripped most (if not all) out in the subviews branch.

**Comment by lynxlynxlynx** [comment]
That's a separate cleanup or bughunting operation, but do note it won't help with #269 caillean, since that's not a text area.

**Comment by burner1024** [comment]
What's the general policy on adding enhancements?
And can they be implemented as plugins or something?
There's a lot of good stuff in [BG2_Improved_GUI](https://github.com/SpellholdStudios/BG2_Improved_GUI), and I might want to try my hand in implementing some (presuming it's mostly python, not c++).

**Comment by lynxlynxlynx** [comment]
More intrusive stuff should use another bit in this `GUIEnhancements` setting. Python doesn't have plugins and while one could maybe emulate that with conditional imports, it would just break sooner or later. Better to have this code in our codebase, so it gets maintenance updates like everything else.

If you tell me which tweaks you're interested in, I can give better guidance.

**Comment by burner1024** [comment]
Most of them, and I have ideas for a few more. Would be great to pick up all low hanging fruits, at least.
But the top picks for me are loot panel, journal quest folding + relocated buttons, and smart off hand slot.

**Comment by lynxlynxlynx** [comment]
- loot panel: ctrl-q does it better, but shouldn't be too hard
- journal quest folding: the hardest of these four, since we have no code for folding or clicking on subparts of a text area
  -  + relocated buttons: this is the only thing that can be done completely in python — if it doesn't work out of the box (if they don't just replace guijrnl.chu)
- smart offhand slot: this sounds quite nasty to implement; I think it would be better to go one step further and support weapon sets like in iwd2. Would only need an updated guiinv.chu (extra slots) and a change to one line in gemrb.ini. Since it requires just data changes, it's better suited for a mod like gemrb-tweaks.

These require only python code changes: Colorize icon of not memorized scrolls , Weapon Damage Range on Character Sheet (similarly for the inventory one, but that also requires a modified chu), Improved Store&Bag Interface, Innate/Ability/Spell Description Screen on Right Click, Additional Character Information on Record Screen

**Comment by bradallred** [comment]
> journal quest folding: the hardest of these four, since we have no code for folding or clicking on subparts of a text area

We actually do have this. Each piece of `Content` can receive clicks, it's how the dialog options work for example. You can add something to `SpanSelector` to nest a `TextContainer` and have the `ValueChange` callback simply set the vertical size of the `TextContainer` to 0. You can even apply an animation to have it gradually collapse/expand.

It might be cleaner to just add an `TextArea::AddContent` method to just insert whatever arbitrary `Content` hierarchy. I'd have to think more on it.

**Comment by lynxlynxlynx** [comment]
Yeah, but exactly that — normal TAs have only one Content.

**Comment by burner1024** [comment]
There's a ctrl-q? Couldn't find any mention of it in the docs.
If it just grabs the nearby loot, I guess it's a good alternative, but not quite the same, as there's a lot of trash that you don't normally need.

Funny, I assumed journal stuff would be the easiest.

As for weapon sets, I'm not familiar with IWD2. I assume those don't allow using the same item in different sets? Then the functionality is not quite the same. Although sets also allow some uses that smart slot can't do.
Ideal UX would be weapon sets allowing shared items, but I'm guessing it would be even harder.

I guess I'll poke around and see if I can get anything easy done. Not a promise, though, for now I just wanted to clarify if these tweaks could be wanted.

**Comment by lynxlynxlynx** [comment]
Ah, mistook it — it's ctrl-w. Mentioned on the innovations page and also as one of the keys in https://gemrb.org/Cheats.html#cheats-and-conveniences
It doesn't grab the loot, just consolidates it into a single pile with the same items merged.

Weapon sets: yes, no sharing. And it would be annoying to implement, so if anyone wanted that, I'd just advise them to cheat in another copy of the weapon for such a purpose.

**Comment by lynxlynxlynx** [comment]
I've added another window in options recently, so this can now be explored and toggled in game. As for the unrelated discussion on BG2-Improved-GUI, there's now a new issue for it.
